src/get_pat.py

```diff
@@ -2,131 +2,272 @@
-		 'response_type=code&client_id={}&redirect_uri={}&' +
-		 'state={}&resource={}')
+# TEMPLATE_AUTHZ_URL = ('https://login.windows.net/{}/oauth2/authorize?' +
+# 		 'response_type=code&client_id={}&redirect_uri={}&' +
+# 		 'state={}&resource={}')
+
 DB_RESOURCE_ID = '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Databricks/workspaces/{}'
 
+LIST_NODE_TYPES_ENDPOINT = '/api/2.0/clusters/list-node-types'
 CREATE_TOKEN_ENDPOINT = '/api/2.0/token/create'
 REDIRECT_URI = 'http://localhost'
 
-def get_authorization_code(tenant_id, client_id, redirect_uri, authorization_url, auth_state):
-	# open browser
-	dr = webdriver.Chrome()
-	# load user login page
-	dr.get(authorization_url)
-	# wait until the user login or kill the process
-	code_received = False
-	code = ''
-
-	while not code_received:
-		cur_url = dr.current_url
-		if cur_url.startswith(redirect_uri):
-			parsed = urlparse(cur_url)
-			query = parse_qs(parsed.query)
-			code = query['code'][0]
-			state = query['state'][0]
-			# throw exception if the state does not match
-			if state != str(auth_state):
-				raise ValueError('state does not match')
-			code_received = True 
-			dr.close()
-
-	if not code_received:
-		log.error('Error in requesting auth code for AAD token')
-		dr.close()
-	# auth code is returned. If not success then empty code returned
-	return code
-
-def get_refresh_and_aad_token(tenant_id, client_id, client_secret, redirect_uri):
-	# build URL for auth code
-	auth_state = uuid.uuid4() # - used for preventing CSRF attacks
-	authorization_url = TEMPLATE_AUTHZ_URL.format(
-		tenant_id,
-		client_id,
-		redirect_uri,
-		auth_state,
-		DATABRICKS_RESOURCE_ID
-	)
-
-	# configure AuthenticationContext (reference link)
-	# authority URL and tenant ID are used
-	authority_url = AUTHORITY_HOST_URL + tenant_id
-	context = AuthenticationContext(authority_url)
-
-	# Obtain the authorization code in by a HTTP request in the browser
-	# then copy it here
-	# Or, call the function above to get the authorization code
-	authz_code = get_authorization_code(tenant_id, client_id, redirect_uri, authorization_url, auth_state)
-	
-	# API call to get the token, the response is a key-value dict
-	token_response = context.acquire_token_with_authorization_code(
-		authz_code,
-		redirect_uri,
-		DATABRICKS_RESOURCE_ID, 
-		client_id,
-		client_secret)
-
-	# you can print all the fields in the token_response
-	for key in token_response.keys():
-		log.debug(str(key) + ': ' + str(token_response[key]))
-	
-	# the tokens can be returned as a pair (or you can return the full
-	# token_response)
-	return (token_response['refreshToken'], token_response['accessToken'])
-
-# Use Token API to exhange AAD token for PAT
-def pat_from_aad_token(aad_token, db_host, subscription_id, rg_name, ws_name):
+
+
+def get_adb_authentication_client_key(params):
+	"""
+    Authenticate using service principal w/ key.
+    """
+	authority_host_uri = 'https://login.microsoftonline.com'
```

Comment:
Can we add this as a global up top so its easier to change later or parameterize if needed

Reply:
done, updated code with requested changes